### ISSUE TITLE
RC1 does not build from VisualStudioOnline Continuous Integration into an Azure Web App

### ISSUE THREAD

**Issue by AlexChesser** [aspnet/dnx#3256]
Apologies if this is the wrong place, with so may inter-related systems it's difficult to figure out exactly where this goes. 

I took a stock RC1 template as created by VisualStudio's new project command.
I set a VSO git code repository
I created an Azure web app and set continuous deployment 
On code checkin - the resulting error is as follows:

```
Issues
Build
C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v14.0\DNX\Microsoft.DNX.targets (119, 0)
The Dnx Runtime package needs to be installed. See output window for more details.
Exception Message: MSBuild error 1 has ended this build. You can find more specific information about the cause of this error in above messages. (type BuildProcessTerminateException)
Exception Stack Trace: at System.Activities.Statements.Throw.Execute(CodeActivityContext context)
at System.Activities.CodeActivity.InternalExecute(ActivityInstance instance, ActivityExecutor executor, BookmarkManager bookmarkManager)
at System.Activities.Runtime.ActivityExecutor.ExecuteActivityWorkItem.ExecuteBody(ActivityExecutor executor, BookmarkManager bookmarkManager, Location resultLocation)
```

Note that I can use a "publish" profile in visual studio and send directly to Azure. From what I gather, there are some specific folders and scripts required to tell a web-app it needs to install the DNX code before this will work.

Perhaps the Azure CI deploy scripts should come part of the stock template?

![capture](https://cloud.githubusercontent.com/assets/355561/11782846/a3859e0e-a241-11e5-8206-56c1a1264e58.PNG)

**Comment by muratg** [comment]
Where do you see this error?  Could you also try to do it from a GitHub repo too (I wonder if there's a difference with the VSO git repo)

**Comment by AlexChesser** [comment]
Hey @muratg the issue has been posted on the MSDN forums where there are a few more details. 

Specifically, it seems that there are specific powershell scripts required to "prepare" an Azure WebApp to be able to compile DNX

There are two blog posts that detail the configurations required. 
https://blogs.msdn.microsoft.com/jamiedalton/2015/10/26/aspnet-5-and-vso-hosted-build-agents-walkthrough/
https://msdn.microsoft.com/Library/vs/alm/Build/azure/deploy-aspnet5?f=255&MSPPError=-2147217396

I have detailed my experiences with this issue (a little) here:

https://social.msdn.microsoft.com/Forums/en-US/75711594-0c39-4597-a90c-737487806383/build-dies-on-deploy-err-cannot-find-path?forum=windowsazurewebsitespreview

Unfortunately, it seems that even through those two posts there are still errors. Likely related to paths or something that can't really be debugged without deeper knowledge of azure or a budget to pay for a large number of tests (each test uses "build minutes" in VSO, so there is a cost associated with fixing this) 

So, the steps to replicate would be:

- create new webproject
- create azure web-app
- link web-app to continuous integration / revision control (any revision control service supported by azure)
- push to revision control

The error occurs in the Azure build.

Specifically, targets have to be installed (sorted via the scripts I've linked) and paths have to be correct to enable the deploy to work properly.
